povm_toolbox/library/dilation_measurement.py

```diff
@@ -0,0 +1,299 @@
+        qc = QuantumCircuit(qr, cr, name="measurement_circuit")
+        for i in range(self.num_qubits):
+            qc.unitary(self._unitary_from_parameters(self._parameters[i]), [i, i + self.num_qubits])
+
+        qc.measure(qr, cr)
+
+        t2 = time.time()
+        LOGGER.info(f"Finished circuit construction. Took {t2 - t1:.6f}s")
+
+        return qc
+
+    def to_sampler_pub(
+        self,
+        circuit: QuantumCircuit,
+        circuit_binding: BindingsArray,
+        shots: int,
+        *,
+        pass_manager: StagedPassManager | None = None,
+    ) -> tuple[SamplerPub, POVMMetadata]:
+        """Append the measurement circuit(s) to the supplied circuit.
+
+        This method takes a supplied circuit and appends the measurement circuit to it.
+        An ancilla qubit is required for each qubit that we want to measure (specified by
+        :attr:`.measurement_layout`). If the input circuit has one or more idling qubit(s),
+        they will be used as measurement ancilla qubits. If more ancilla qubits are needed,
+        those will be added to the input circuit (therefore increasing its size).
+
+        .. warning::
+           The number of qubits in the input circuit may be increased due to the ancilla
+           qubits qubits required for dilatation measurements.
+
+        Args:
+            circuit: A quantum circuit.
+            circuit_binding: A bindings array.
+            shots: A specific number of shots to run with.
+            pass_manager: An optional transpilation pass manager. After the supplied circuit has
+                been composed with the measurement circuit, the pass manager will be used to
+                transpile the composed circuit.
+
+        Returns:
+            A tuple of a sampler pub and a dictionary of metadata which includes the
+            :class:`.POVMImplementation` object itself. The metadata should contain all the
+            information necessary to extract the POVM outcomes out of raw bitstrings.
+        """
+        t1 = time.time()
+        LOGGER.info("Piecing together SamplerPub")
+
+        composed_circuit = self.compose_circuits(circuit)
+
+        if pass_manager is not None:
+            composed_circuit = pass_manager.run(composed_circuit)
+
+        pub = SamplerPub(
+            circuit=composed_circuit,
+            parameter_values=circuit_binding,
+            shots=shots,
+        )
+
+        metadata = POVMMetadata(povm_implementation=self, composed_circuit=composed_circuit)
+
+        t2 = time.time()
+        LOGGER.info(f"Finished building SamplerPub. Took {t2 - t1:.6f}s")
+
+        return (pub, metadata)
+
+    @override
+    def reshape_data_bin(self, data: DataBin) -> DataBin:
+        return data
+
+    @override
+    def _povm_outcomes(
+        self,
+        bit_array: BitArray,
+        povm_metadata: POVMMetadata,
+        loc: int | tuple[int, ...] | None = None,
```

Comment:
```suggestion
        povm_metadata: POVMMetadata,
        *,
        loc: int | tuple[int, ...] | None = None,
```
Can you please propagate this also to all other `_povm_outcomes`  methods in the other classes?